sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ClassName.cs

```diff
@@ -9,6 +9,11 @@ static class IEnumerableExtensions // Compliant
     {
     }
 
+    // https://github.com/SonarSource/sonar-dotnet/issues/3268
+    class IEnumerableExtensionsTests // Noncompliant FP, Classes ending with "Test" or "Tests" should not be reported
```

Comment:
it seems to me the fix for this is low-hanging, so I suggest to make it in this PR when getting some time. WDYT?

https://github.com/SonarSource/sonar-dotnet/blob/master/sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/Naming/ClassName.cs#L53

Reply:
also, this FP applies to both C# and VB.NET, right?

Reply:
No, it's not related to VB.NET. VB.NET has very simplified implementation (regex-check-only) with no other logic for now.

Reply:
It is low hanging indeed, just new if..return somewhere here in C# rule:
[https://github.com/SonarSource/sonar-dotnet/blob/3be496ff5f8d01ca638eee1a8d4411dc8c55439d/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ClassAndMethodName.cs#L112](https://github.com/SonarSource/sonar-dotnet/blob/3be496ff5f8d01ca638eee1a8d4411dc8c55439d/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ClassAndMethodName.cs#L112)